### DISCUSSION TITLE
IPv6 support

### DISCUSSION THREAD

**Discussion by steffann** [liqotech/liqo#1378] [General]
I am trying to use liqo in a dual-stack cluster, but liqoctl only seems to accept IPv6 addresses for --pod-cidr and --service-cidr. Am I doing something wrong or is IPv6 just not supported?

If it's not supported, what would it take to get it supported?

**Comment by giorio94** [comment]
Hi @steffann,

Currently, liqo does not support IPv6 networks and dual-stack clusters. 

I feel it should be feasible to extend the current approach to IPv6 addresses too (exchange of CIDRs between the clusters, possible remapping, definition and creation of the appropriate routing/iptables rules), although that would require a quite significant amount of work to appropriately modify the different components. In case you would like to contribute in that direction, we could analyze more in depth the necessary steps towards the goal.

**Comment by kisg** [comment]
Hi @giorio94,

I just found Liqo a couple of days ago. I started researching it for a complex, multi-tenant use case. We plan to use IPv6 exclusively to have clearly defined address spaces for each tenant and each edge cluster associated with a tenant.

Is IPv6 support planned or would a contribution to support it be accepted?

Kind regards,
Gergely

**Reply by giorio94** [reply]
Hi @kisg,

Thanks for your interest in Liqo! I'll defer this to @aleoli and @cheina97 for more info about the planned directions and guidance for a possible contribution.

**Comment by cheina97** [comment]
Hi @kisg,
 
Unfortunately, Liqo does not support **ipv6** and is not part of the roadmap.
Anyway, at the moment we are working on the separation between Liqo and its **network-fabric** (#1671 ), which will allow using third-party solutions like submariner or cilium-multicluster. 

Any contribution to the project would be greatly appreciated. A starting point could be your analysis of what is "breaking" using Liqo in your scenario and what could be the causes.
